ACPs/209-eip7702-style-account-abstraction/README.md

```diff
@@ -0,0 +1,127 @@
+
+This ACP proposes adding a similar transaction type and functionality to Avalanche EVM implementations in order to have them support the same style of UX available on Ethereum. Modifications to the handling of account nonce and balances are required in order for it to be safe when used in conjunction with the streaming asynchronous execution (SAE) mechanism proposed in [ACP-194](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/194-streaming-asynchronous-execution).
+
+## Motivation
+
+The motivation for this ACP is the same as the motivation described in [EIP-7702](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md#motivation). However, EIP-7702 as implemented for Ethereum breaks invariants required for EVM chains that use the ACP-194 SAE mechanism.
+
+There has been strong community feedback in support of ACP-194 for its potential to:
+- Allow for increasing the target gas rate of Avalanche EVM chains, including the C-Chain
+- Enable the use of an encrypted mempool to prevent front-running
+- Enable the use of real time VRF during transaction execution
+
+Given the strong support for ACP-194, bringing EIP-7702-style functionality to Avalanche EVMs requires modifications to preserve its necessary invariants, described below.
+
+### Invariants needed for ACP-194
+
+There are [two invariants explicitly broken by EIP-7702](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md#backwards-compatibility) that are required for SAE. They are:
+
+1. An account balance can only decrease as a result of a transaction originating from that account.
+1. An EOA nonce may not increase after transaction execution has begun.
+
+These invariants are required for SAE in order to be able to statically analyze (i.e. determine without executing the transaction) that a transaction:
+- Has the proper nonce
+- Will have sufficient balance to pay for its worst case transaction fee plus the balance it sends
+
+As described in the ACP-194, this lightweight analysis of transactions in blocks allows blocks to be accepted by consensus with the guarantee that they can be executed successfully. Only after block acceptance are the transactions within the block then put into a queue to be executed asynchronously. If the execution of transactions in the queue can decrease an EOA's account balance or change an EOA's current nonce, then block verification is unable to ensure that transactions in the block will be valid when executed. If transactions accepted into blocks can be invalidated prior to their execution, this poses DOS vulnerabilities because the invalidated transactions use up resources but do not pay any fees.
```

Comment:
"use up resources but do not pay any fees" - can we specify what resources they consume? We should be specific and reference space on the queue as opposed to actual execution resources, since we should be able to throw it out at execution time

Reply:
Great call. Reworded to "...because the invalidated transactions use up space in the pending execution queue according to their gas limits, but they do not pay any fees."